src/english_score_eda.py

```diff
@@ -0,0 +1,186 @@
+        alt.Chart(train_df)
+        .mark_bar()
+        .encode(
+            x=alt.X("education:N", title="Education Level"),
+            y=alt.Y("count():Q", title="Count"),
+            tooltip=["count()"],
+        )
+        .properties(title="Distribution of Education Levels")
+    )
+
+    # Save the education level plot
+    education_level_plot.save(plot_to + '/education-level-fig.png', format='png')
+
+    # Categorical Feature Plot
+    categorical_plot = alt.Chart(train_df, title="Distribution of the Categorical features").mark_bar().encode(
+        x=alt.X(alt.repeat()),
+        y="count()",
+    ).properties(height=200, width=800).repeat(categorical_feats, columns=1)
+
+    # Save the categorical plots
+    categorical_plot.save(plot_to + '/feat-categoric-figs.png', format='png')
+
+    # Columns for the correlation matrix
+    normal_cols = list(
+        set(train_df.columns.tolist())
+        - set(
+            [col for col in train_df.columns if col.startswith("q")]
+            + ["Unnamed: 0", "date", "time", "id"]
+        )
+    )
+    
+    # Generate and save the Pearson correlation matrix
+    cm = pearson_correlation_matrix(train_df[normal_cols], "seismic")
+    dfi.export(cm, plot_to+'feat-correlation-matrix.png')
+    
+
+    # Final additional feature definitions 
+    numeric_feats = ["age", "Eng_start", "Eng_country_yrs", "Lived_Eng_per"]
+
+    binary_feats = ["psychiatric"]
+    target = ["correct"]
+    binary_withNA = [
+    "house_Eng",
+    "nat_Eng",
+    "prime_Eng",
+    ] 
+
+    numeric_transformer = make_pipeline(
+    SimpleImputer(strategy="median"), StandardScaler())
+
+    categorical_transformer = make_pipeline(
+        SimpleImputer(strategy="constant", fill_value="little"),
+        OneHotEncoder(handle_unknown="ignore"),
+    )
+
+    binary_NA_transformer = make_pipeline(
+        SimpleImputer(strategy="constant", fill_value=0),
+        OneHotEncoder(
+            handle_unknown="ignore" ,drop="if_binary", dtype=int
+        ),
+    )
+
+    # Get value counts for the 'education' column
+    education_counts = train_df["education"].value_counts()
+
+    # Filter to include only counts greater than 1
+    education_counts_greater_than_one = education_counts[education_counts > 100]
+
+    categories_list = education_counts_greater_than_one.index.tolist()
+
+
+    
+
+
+    # Defines the order for Education to be used in the OrdinalEncoder
+    education_order = [
+        "Graduate Degree",
+        "Some Graduate School",
+        "Undergraduate Degree (3-5 years higher ed)",
+        "Some Undergrad (higher ed)",
+        "High School Degree (12-13 years)",
+        "Haven't Finished High School (less than 13 years ed)",
+        "Didn't Finish High School (less than 13 years ed)",
+        "Others",
+    ]
+
+
+    # Create a transformer using FunctionTransformer
+    categorical_education_tranformer = make_pipeline(
+        FunctionTransformer(map_to_other), OrdinalEncoder(
```

Comment:
I think, like in the change I made earlier this week, wee need to add feature_names_out='one-to-one' for us to be able to use .get_feature_names_out() later in our function

Reply:
Yep, can confirm this is needed